src/mongo/bson/bsonelement.cpp

```diff
@@ -145,8 +145,9 @@ string BSONElement::jsonString(JsonStringFormat format, bool includeFieldNames,
                         for (int x = 0; x < pretty; x++)
                             s << "  ";
                     }
-
-                    if (strtol(e.fieldName(), 0, 10) > count) {
+                    long int field_digit = 0;
+                    if (parseNumberFromString<long int>(e.fieldName(), &field_digit) == Status::OK()
```

Comment:
We typically use the isOK() method on Status to check if it's OK rather than comparing directly to Status::OK(). For consistency with other code in our codebase, please do that in this patch.